ambry-api/src/main/java/com.github.ambry/store/Store.java

```diff
@@ -71,6 +71,13 @@
   Set<StoreKey> findMissingKeys(List<StoreKey> keys) throws StoreException;
 
   /**
+   * Get stats related information about the store as a {@link StatsDirectory}.
+   * @return a {@link StatsDirectory} containing stats related information
+   * @throws StoreException
+   */
+  StatsDirectory getStoreStats() throws StoreException;
```

Comment:
`getStatsSnapshot()` maybe better since there is an interface named `StoreStats` and `getStoreStats()` should probably be a function that gets an instance that implements `StoreStats`

Also, should this function be in `StoreStats` rather than in `Store`? This interface can probably have `StoreStats getStoreStats()` and `StoreStats` can have `StatsSnapshot getStatsSnapshot()`.